torch/distributed/distributed_c10d.py

```diff
@@ -598,7 +598,7 @@ def isend(tensor,
         tag (int, optional): Tag to match send with remote recv
 
     Returns:
-        A distributed request object.
+        Async work handle, call wait() on the handle to wait until it finishes.
```

Comment:
I have a nit comment, but feel free to ignore if you think it doesn't help ;) 
I wonder if it worth saying `returns a torch.distributed.Work` object here and add documentation for `torch.distributed.Work`? In that case we don't have to duplicate `call wait() on the handle to wait until it finishes` and users could find that out themselves. (It's also easier if we want to change/add something to this object in the future.

Reply:
Yep, that would make the doc look better

@pietern is it OK that we expose the `torch.distributed.Work` API in the doc?

Reply:
Yep, sure, let's do it. For asynchronous operations that operate on new tensors we will also have to include how you can retrieve the resulting tensor, which we don't do today.